Calculator/Calculator/Program.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+namespace Calculator
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int previous = default;
+            int current = default;
+            string operation = default;
+
+            while(true)
+            {
+                var askFirstOperand = true;
+                var askSecondOperand = true;
+                var askOperation = true;
+                
+                while(askFirstOperand)
+                {
+                    Console.Write("Введите число: ");
+                    var priv = Console.ReadLine();
+                    try
+                    {
+                        previous = int.Parse(priv);
+                        askFirstOperand = false;
+                    }
+                    catch (Exception)
+                    {
+                        Console.WriteLine("Это не число");
+                    }
+                }
+                
+                while(askOperation)
+                {
+                    Console.Write("Введите операцию: ");
+                    operation = Console.ReadLine();
+                    
+                    if (operation == "!")
+                    {
+			askOperation = false;
+			askSecondOperand = false;
+                    }
+                    else if (operation == "+" || operation == "-" || operation == "*" || operation == "/" || operation == "^")
+                    {
+                        askOperation = false;
+                    }
+                    else
+                    {
+                        Console.WriteLine("Неверная операция, доступны только: + - * / ^ !");
```

Comment:
User think; But i didn`t know about that restrictions before....

Reply:
Fixed in commit abc5bfa3e7ef4244382d590cbb9db7f391f18fa2